ee/vellum_ee/workflows/display/tests/workflow_serialization/test_basic_inline_prompt_node_serialization.py

```diff
@@ -274,3 +279,44 @@ def test_serialize_workflow():
         "name": "BasicInlinePromptWithFunctionsWorkflow",
         "module": ["tests", "workflows", "basic_inline_prompt_node_with_functions", "workflow"],
     }
+
+
+def test_serialize_workflow_with_descriptor_functions():
+    """Test that serialization handles BaseDescriptor instances in functions list."""
+
+    def example_function():
+        """Example function for testing."""
+        return "example"
+
+    class TestInputs(BaseInputs):
+        noun: str
+
+    class MockMCPClientNode:
+        class Outputs:
+            tools = [example_function, FunctionDefinition(name="test", description="test")]
+
+    class TestInlinePromptNodeWithDescriptorFunctions(InlinePromptNode):
+        ml_model = "gpt-4o"
+        blocks = [
+            ChatMessagePromptBlock(
+                chat_role="SYSTEM",
+                blocks=[JinjaPromptBlock(template="Test {{noun}}")],
+            ),
+        ]
+        prompt_inputs = {"noun": TestInputs.noun}
+        functions = MockMCPClientNode.Outputs.tools
+
+    class TestWorkflow(BaseWorkflow[TestInputs, BaseState]):
+        graph = TestInlinePromptNodeWithDescriptorFunctions
+
+    workflow_display = get_workflow_display(workflow_class=TestWorkflow)
+    serialized = workflow_display.serialize()
+
+    prompt_nodes = [node for node in serialized["workflow_raw_data"]["nodes"] if node["type"] == "PROMPT"]
+    assert len(prompt_nodes) == 1
+
+    prompt_node = prompt_nodes[0]
+    blocks = prompt_node["data"]["exec_config"]["prompt_template_block_data"]["blocks"]
+
+    function_blocks = [block for block in blocks if block.get("block_type") == "FUNCTION_DEFINITION"]
+    assert len(function_blocks) == 2  # Should have 2 function blocks for the example_function and FunctionDefinition
```

Comment:
no, this should now be 0, we skip prompt block serialization in this dynamic case